src/mlpack/tests/init_rules_test.cpp

```diff
@@ -123,4 +126,38 @@ BOOST_AUTO_TEST_CASE(OivsInitTest)
   BOOST_REQUIRE_EQUAL(1, 1);
 }
 
+// Test the GaussianInitialization class.
+BOOST_AUTO_TEST_CASE(GaussianInitTest)
+{
+  arma::mat weights;
+  arma::cube weights3d;
+  const size_t row = 7;
+  const size_t col = 7;
+  const size_t slice = 2;
+  double mean, mean3d;
+  double var,var3d;
+  mean=mean3d=1;
+  var=var3d=1;
+  GaussianInitialization t(0, 0.2);
+
+  for(size_t i =0; i<10; i++)
+  {
+    t.Initialize(weights, row, col);
+    t.Initialize(weights3d, row, col, slice);
+    distribution::GaussianDistribution Dist;
+    mean += arma::accu(weights) / weights.n_elem;
+    var += arma::accu(pow((weights.t() - mean), 2)) / weights.n_elem -1 ;
+    mean3d += arma::accu(weights3d.slice(0)) / weights3d.slice(0).n_elem;
+    var3d += arma::accu(pow((weights3d.slice(0) - mean), 2)) / weights3d.slice(0).n_elem -1;
+  }
+  mean = mean/10;
+  var = var/10;
+  mean3d /= 10;
+  var3d /= 10;
+  BOOST_REQUIRE_SMALL(mean-0, 0.5);
```

Comment:
Maybe we can choose a better threshold here, since what we do here is to test if e.g. ``mean < 0.5``.